File: leave-approval-frontend/src/services/adminUserService.js
// src/services/adminUserService.js
import api from './api'; // 你的 Axios 实例

class AdminUserService {
  /**
   * 获取所有用户列表 (分页)
   * @param {object} params - 分页和排序参数 (e.g., { page: 0, size: 10, sort: 'id,asc' })
   * @returns {Promise<object>} - 包含分页信息的用户列表 (Page<UserDto>)
   */
  async getAllUsers(params = {}) {
    try {
      const response = await api.get('/admin/users', { params });
      return response.data;
    } catch (error) {
      console.error('Error fetching all users:', error);
      throw error;
    }
  }

  /**
   * 管理员创建新用户
   * @param {object} userData - 用户数据 (AdminUserCreateRequest DTO)
   * @returns {Promise<object>} - 创建的用户信息 (UserDto)
   */
  async createUser(userData) {
    try {
      const response = await api.post('/admin/users', userData);
      return response.data;
    } catch (error) { // <--- 修改点：添加了开始的花括号 {
      console.error('Error creating user:', error);
      throw error;
    } // <--- 这个是 catch 块的结束花括号
  }

  /**
   * 管理员更新用户信息
   * @param {number} userId - 用户ID
   * @param {object} userData - 要更新的用户数据 (UserUpdateRequest DTO)
   * @returns {Promise<object>} - 更新后的用户信息 (UserDto)
   */
  async updateUser(userId, userData) {
    try {
      const response = await api.put(`/admin/users/${userId}`, userData);
      return response.data;
    } catch (error) {
      console.error(`Error updating user ${userId}:`, error);
      throw error;
    }
  }

  /**
   * 管理员删除用户
   * @param {number} userId - 用户ID
   * @returns {Promise<object>} - 删除成功的消息 (MessageResponse)
   */
  async deleteUser(userId) {
    try {
      const response = await api.delete(`/admin/users/${userId}`);
      return response.data;
    } catch (error) {
      console.error(`Error deleting user ${userId}:`, error);
      throw error;
    }
  }

  /**
   * 管理员根据ID获取单个用户信息
   * @param {number} userId - 用户ID
   * @returns {Promise<object>} - 用户信息 (UserDto)
   */
  async getUserById(userId) {
     try {
        const response = await api.get(`/admin/users/${userId}`);
        return response.data;
     } catch (error) {
        console.error(`Error fetching user ${userId}:`, error);
        throw error;
     }
  }

  // --- 新增方法：获取可选的经理列表 ---
  /**
   * 获取可作为经理的用户列表。
   * @returns {Promise<Array<object>>} - 一个包含用户DTO的数组，每个对象至少有 id, fullName, username, roles。
   */
  async getPotentialManagers() {
    try {
      const response = await api.get('/auth/potential-managers');
      return response.data;
    } catch (error) {
      console.error('AdminUserService: Error fetching potential managers:', error);
      throw error;
    }
  }
  // --- 新增方法结束 ---

  // 你可以在这里添加获取所有可用角色的方法，如果后端提供了API
  // async getAvailableRoles() { ... }
}

export default new AdminUserService();

File: leave-approval-backend/src/main/java/com/example/leaveapproval/service/user/AdminUserService.java
package com.example.leaveapproval.service.user;

import com.example.leaveapproval.dto.AdminUserCreateRequest;
import com.example.leaveapproval.dto.UserDto;
import com.example.leaveapproval.dto.UserUpdateRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List; // 新增：导入 List
import java.util.Optional;

/**
 * 用户管理服务接口 (管理员权限)。
 * 定义了管理员对用户实体进行操作的业务逻辑契约。
 */
public interface AdminUserService {

    /**
     * 分页获取所有用户信息。
     *
     * @param pageable 分页参数对象。
     * @return 包含用户DTO的分页结果对象。
     */
    Page<UserDto> getAllUsers(Pageable pageable);

    /**
     * 根据用户ID获取用户信息。
     *
     * @param id 用户的唯一标识ID。
     * @return 包含用户DTO的Optional对象，如果用户不存在则为空。
     */
    Optional<UserDto> getUserById(Long id);

    /**
     * 管理员创建新用户。
     *
     * @param createRequest 包含新用户创建信息的DTO对象。
     * @return 创建成功后的用户DTO对象。
     * @throws org.springframework.dao.DataIntegrityViolationException 如果用户名或邮箱已存在。
     * @throws IllegalArgumentException 如果DTO中未指定用户角色。
     */
    UserDto createUser(AdminUserCreateRequest createRequest);

    /**
     * 管理员更新指定ID的用户信息。
     *
     * @param id 要更新的用户的ID。
     * @param updateRequest 包含用户更新信息的DTO对象。
     * @return 更新成功后的用户DTO对象的Optional，如果用户不存在则为空。
     * @throws com.example.leaveapproval.exception.ResourceNotFoundException 如果用户或其指定的上级用户不存在。
     * @throws org.springframework.dao.DataIntegrityViolationException 如果更新后的邮箱与现有其他用户冲突。
     * @throws IllegalArgumentException 如果尝试将用户角色列表更新为空。
     */
    Optional<UserDto> updateUser(Long id, UserUpdateRequest updateRequest);

    /**
     * 管理员根据ID删除用户。
     *
     * @param id 要删除的用户的ID。
     * @throws com.example.leaveapproval.exception.ResourceNotFoundException 如果指定ID的用户不存在。
     */
    void deleteUser(Long id);

    // --- 新增方法签名 ---
    /**
     * 获取可作为经理的用户列表。
     * 这些用户通常拥有特定的管理角色 (例如 TeamLead, DeptManager, HR, Admin)。
     * 用于前端下拉列表选择经理。
     *
     * @return 可选经理的用户DTO列表。
     */
    List<UserDto> getPotentialManagers();
    // --- 新增方法签名结束 ---
}